### ISSUE TITLE
Fields cannot be found after using eval keep with named fields

### ISSUE THREAD

**Issue by jdursel** [influxdata/kapacitor#1026]
Hi,

Some fields from previous nodes are disapearing when using eval keep with fields parameters

here's the script which is not working because field ppos cannot be found on last keep
```
 var price = batch
     |query('''SELECT last("ppos") as "ppos" , last("pneg") as "pneg" FROM die."default".ImbalancePrice''')
         .cron(cronDefault)
         .period(45m)
     |shift(45m)
     |eval(lambda: 15.0 * trunc(float(hour("time") * 60 + minute("time")) / 15.0))
         .as('minutesEnd')
         .keep()
     |eval(lambda: mod("minutesEnd", 60.0))
         .as('minuteEnd')
         .keep()
     |eval(lambda: trunc("minutesEnd" / 60.0))
         .as('hourEnd')
         .keep()
     |eval(lambda: mod("minutesEnd" - 15.0, 1440.0))
         .as('minutesStart')
         .keep()
     |eval(lambda: mod("minutesStart", 60.0))
         .as('minuteStart')
         .keep()
     |eval(lambda: trunc("minutesStart" / 60.0))
         .as('hourStart')
         .keep('hourStart', 'minuteStart', 'hourEnd', 'minuteEnd', 'ppos', 'pneg')
     |influxDBOut()
         .database('tpm_die')
         .retentionPolicy('default')
         .measurement('PriceAlert')
         .precision('m')
```

To make it work, it is mandatory to either remove all named fields from the keep (obviously leaving some unnecessary fields in the payload for influxdb) or to add the fields in lambdas, "as" and "keep":
```    |eval(lambda: trunc("minutesStart" / 60.0), lambda: "minuteStart", lambda: "hourEnd", lambda: "minuteEnd", lambda: "ppos", lambda: "pneg")
        .as('hourStart', 'minuteStart', 'hourEnd', 'minuteEnd', 'ppos', 'pneg')
        .keep('hourStart', 'minuteStart', 'hourEnd', 'minuteEnd', 'ppos', 'pneg')
```
It seems that the fields to be kept must be mentionned in the "as" (and therefore in the lambdas) to be kept by the "keep" property.

Is it a regression bug or does it work like that now?
Thank you.

John.

**Comment by jdursel** [comment]
version 1.0.2
May be linked to #984

**Comment by nathanielc** [comment]
@jdursel #984 was fixed in version 1.1. I have tested your script with 1.1 and the issue seems to be resolved. Can you confirm?

**Comment by jdursel** [comment]
I confirm, the issue is resolved in version 1.1.

thanks
